flake8_trio/visitors/helpers.py

```diff
@@ -125,6 +142,62 @@ def iter_guaranteed_once(iterable: ast.expr) -> bool:
     return False
 
 
+def cst_literal_eval(node: cst.BaseExpression) -> Any:
+    ast_node = cst.Module([cst.SimpleStatementLine([cst.Expr(node)])]).code
+    try:
+        return ast.literal_eval(ast_node)
+    except Exception:  # noqa: PIE786
+        return None
```

Comment:
major hoops to have to run through, but didn't find any better way of doing it for now.

Reply:
Oh, in the case of `iter_guaranteed_once` (literal parameter to range) it's actually plenty fine to use Integer/Float.evaluated_value; but I opened https://github.com/Instagram/LibCST/issues/878 for other cases.